### ISSUE TITLE
New generics

### ISSUE THREAD

**Issue by lgatto** [rformassspectrometry/ProtGenerics#8]
Generics to be moved here (from `MSnbase`)
- ~`smooth`~
- ~`centroided` getters and setters~
- ~`instrumentModel`~
- ~`processingData`~

**Comment by kuwisdelu** [comment]
Additional setters for consideration (from `Cardinal`)

- ~`peaks<-`~
- ~`spectra<-`~

I also internally use (and currently export) a setter for:

- ~`processingData<-`~

I use for updating a list of queued processing steps for delayed execution, which I do want users to be able to manipulate.

**Comment by lgatto** [comment]
@kuwisdelu - these (and others) have now been added to `ProtGenerics` 1.17.1 (currently only available on github). ~Will push to Bioc asap.~ Pushed to Bioc now.

**Comment by jorainer** [comment]
Would propose also these:
- [x] `isolationWindowTargetMz`, `isolationWindowTargetMz<-`
- [x] `isolationWindowLowerMz`, `isolationWindowLowerMz<-`
- [x] `isolationWindowUpperMz`, `isolationWindowUpperMz<-`
- [x] `intensity<-`
- [x] `precursorMz<-`
- [x] `precursorIntensity<-`
- [x] `msLevel<-`
- [x] `collisionEnergy<-`
- [x] `polarity<-`
- [x] `rtime<-`
- [x] `spectraNames`, `spectraNames<-`
- [x] `spectraData`, `spectraData<-`
- [x] `spectraVariables`

**Comment by lgatto** [comment]
@jorainer, @sgibb, @kuwisdelu: do we agree to use the following signature in all (or most) cases?

```r
setGeneric("", function(object, ...) standardGeneric(""))
setGeneric("<-", function(object, ..., value) standardGeneric("<-"))
```

@jorainer - as for all the accessors, having the generics doesn't imply anything, but I am not sure actually implementing them (making them available to users) is always useful or necessary. Some slots aren't necessarily meant to be changed.

**Comment by kuwisdelu** [comment]
Those signatures are fine with me, yes.

**Comment by jorainer** [comment]
yes to the signature.

Agree, not all need to be implemented. But some variables I need to be changeable, such as `rtime<-` (for alignment/retention time adjustment), `isolationWindowTargetMz` ... (for DIA/SWATH analysis).

**Comment by lgatto** [comment]
I have just pushed `ProtGenerics` 1.17.2 to Bioc that has some additional generic definitions and exports all of them (which I had forgotten to do in 1.17.1), so please update once the latest one is available and depend on >= 1.17.2 rather that 1.17.1.
